File: BOJ(Java 공부용)/11652.java
import java.io.*;
import java.util.*;

public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static int[] gox = {0, 1, -1, 0, 1, 1, -1, -1};
    static int[] goy = {1, 0, 0, -1, 1, -1, 1, -1};

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        TreeMap<Long, Integer> m = new TreeMap<>();
        for (int i = 0; i < n; i++) {
            long num = Long.parseLong(br.readLine());
            if (m.get(num) == null) {
                m.put(num, 1);
            } else {
                int value = m.get(num);
                value++;
                m.remove(num);
                m.put(num, value);
            }
        }

        Iterator it = m.keySet().iterator();
        int max = 0;
        long temp = 0;
        while (it.hasNext()) {
            long key = (long) it.next();
            int value = m.get(key);
            if (max < value) {
                max = value;
                temp = key;
            }
        }
        System.out.println(temp);
    }

}

File: BOJ(Kotlin)/11652.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

fun main(args : Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    var st = StringTokenizer(br.readLine())
    val n = st.nextToken().toInt()
    val treeMap = TreeMap<Long,Int>()
    for (i in 0 .. n-1) {
        val num : Long = br.readLine().toLong()
        if (treeMap[num] == null) {
            treeMap[num] = 1
        }
        else {
            val value  = treeMap[num]
            treeMap.remove(num)
            if (value != null) {
                treeMap[num] = value + 1
            }
        }
    }

    val it = treeMap.iterator()
    var max = 0
    var temp : Long = 0
    while(it.hasNext()) {
        var key : Long = it.next().key
        val value = treeMap[key]
        if (max < value!!) {
            max = value
            temp = key
        }
    }
    bw.write(temp.toString())
    bw.flush()
    bw.close()
}
